Add `.multiple()` method

Richienb opened the pull request:
Fixes: #3

Richienb committed:
Add `.multiple()` method

Signed-off-by: Richie Bendall <richiebendall@gmail.com>

sindresorhus commented on index.js:
```diff
@@ -7,3 +7,21 @@ module.exports = array => {
+
+	array = [...array];
+
+	return [...new Array(count)].map(() => array.splice(Math.floor(Math.random() * array.length), 1)[0]);
```
Seems like you can do this without splicing (which is very slow).

sindresorhus replied:
Any reason you're not just calling `module.exports()` multiple times?

Richienb replied:
@sindresorhus Which type of behavior do you want?

Allowing duplicates:

```js
randomItem.multiple([1, 2, 3], 2)
//=> [1, 1]
```

Disallowing duplicates:

```js
randomItem.multiple([1, 2, 3], 2)
//=> [2, 1]
```

sindresorhus replied:
Allowing duplicates. That would be consistent with the main method. There could potentially be an option to ensure the elements are consecutively unique, and one that ensures the array uses up all elements before starting again.

sindresorhus replied:
The behavior should be clearly documented.

sindresorhus commented on index.js:
```diff
@@ -7,3 +7,21 @@ module.exports = array => {
+	}
+
+	if (count > array.length) {
+		throw new TypeError(`Expected an array with at least ${count} items`);
```
This should be allowed.

Richienb committed:
Allow specifying more items than in the array

Richienb committed:
Allow duplicate items

Signed-off-by: Richie Bendall <richiebendall@gmail.com>

Richienb committed:
Remove moot check

The check is already done when `randomItem()` is called so calling it again would be unnessesary.

Signed-off-by: Richie Bendall <richiebendall@gmail.com>

Richienb committed:
Merge branch 'add-multiple' of https://github.com/Richienb/random-item into add-multiple

sindresorhus commented:
Any thoughts on https://github.com/sindresorhus/random-item/pull/4#discussion_r511670522 ?

Richienb commented:
@sindresorhus We could add `.multiple` for https://github.com/sindresorhus/unique-random
